testutils/relay.go

```diff
@@ -84,8 +85,14 @@ func (r *frameRelay) relayConn(c net.Conn) {
 		return
 	}
 	r.Lock()
+	defer r.Unlock()
+
+	if r.closed.Load() > 0 {
+		outC.Close()
+		return
+	}
+
```

Comment:
We're losing track of the rebases here...this diff doesn't include the changes from https://github.com/uber/tchannel-go/pull/333/commits/6017cf77af9468f6d199ae173ca5130cd1907c8c.